source/MaterialXView/Material.cpp

```diff
@@ -288,16 +288,16 @@ void Material::bindViewInformation(const mx::Matrix44& world, const mx::Matrix44
     mx::Matrix44 invTransWorld = world.getInverse().getTranspose();
 
     // Bind view properties.
-    _glShader->setUniform(mx::HW::WORLD_MATRIX, ng::Matrix4f(world.getTranspose().data()));
-    _glShader->setUniform(mx::HW::VIEW_PROJECTION_MATRIX, ng::Matrix4f(viewProj.getTranspose().data()));
+    _glShader->setUniform(mx::HW::WORLD_MATRIX, ng::Matrix4f(world.getTranspose().data()), _enableWarnings);
```

Comment:
Instead of an enable warnings flag, it may be better just to check for the presence of each uniform before we set it.  You can see that pattern below for the WORLD_INVERSE_TRANSPOSE_MATRIX uniform, which is not always present when this method is called.

Reply:
Agreed.